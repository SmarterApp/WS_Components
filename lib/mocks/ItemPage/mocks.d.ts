import { match } from "react-router";
import { ItemViewerContainerProps, ItemModel, ItemPageModel, ItemIsaapModel, AccResourceGroupModel, AboutItemModel } from "@src/index";
export declare const aboutThisClient: (params: ItemModel) => Promise<AboutItemModel>;
export declare const itemPageClient: (params: ItemModel) => Promise<ItemPageModel>;
export declare const itemAccessibilityClient: (params: ItemIsaapModel) => Promise<AccResourceGroupModel[]>;
export declare const itemPagePath = "/:bankKey:itemKey";
export declare const itemPageMatch: match<ItemModel>;
export declare const onSave: (accGRoups: AccResourceGroupModel[]) => void;
export declare const onReset: () => void;
export declare const itemPageModelMock: ItemPageModel;
export declare const itemPageMockProps: ItemViewerContainerProps;
export declare const itemPageMockPropsNoItem: ItemViewerContainerProps;
