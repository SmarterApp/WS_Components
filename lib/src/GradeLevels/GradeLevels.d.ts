export declare enum GradeLevels {
    NA = 0,
    Grade3 = 1,
    Grade4 = 2,
    Grade5 = 4,
    Grade6 = 8,
    Grade7 = 16,
    Grade8 = 32,
    Grade9 = 64,
    Grade10 = 128,
    Grade11 = 256,
    Grade12 = 512,
    Elementary = 7,
    Middle = 56,
    High = 960,
    All = 1023,
}
export declare class GradeLevel {
    static gradeCaseToString(grade: GradeLevels): string | undefined;
    static gradeCaseToShortString(grade: GradeLevels): string | undefined;
    static gradeLevelToString(grades: GradeLevels): string | undefined;
    static gradeLevelContains(haystack: GradeLevels, needle: GradeLevels): boolean;
    static stringToGradeLevel(gradeString: string): GradeLevels;
    static gradeLevelAdd(grade: GradeLevels, gradeString: string): number;
}
