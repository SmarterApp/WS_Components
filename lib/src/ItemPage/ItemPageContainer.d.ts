import * as React from "react";
import { AccResourceGroupModel, AboutItemModel, ItemModel, ItemIsaapModel, ItemPageModel, ItemIdentifierModel, Resource } from "@src/index";
export interface ItemPageContainerProps {
    aboutThisClient: (params: ItemModel) => Promise<AboutItemModel>;
    itemPageClient: (params: ItemModel) => Promise<ItemPageModel>;
    itemAccessibilityClient: (params: ItemIsaapModel) => Promise<AccResourceGroupModel[]>;
    showRubrics: boolean;
    itemIsaap: ItemIsaapModel;
    updateIsaap: (isaap: string) => void;
    updateCookie: (cookieName: string, cookieValue: string) => void;
    errorRedirectPath: string;
}
export interface ItemPageContainerState {
    aboutThisItem: Resource<AboutItemModel>;
    itemPageVM: Resource<ItemPageModel>;
    itemAccessibility: Resource<AccResourceGroupModel[]>;
    currentItem?: ItemIdentifierModel;
    item: ItemModel;
    loading: boolean;
    redirect: boolean;
}
export declare class ItemPageContainer extends React.Component<ItemPageContainerProps, ItemPageContainerState> {
    private subscription;
    constructor(props: ItemPageContainerProps);
    componentDidMount(): void;
    componentWillUnmount(): void;
    fetchItemPage(item: ItemModel): void;
    componentWillReceiveProps(nextProps: ItemPageContainerProps): void;
    private setCurrentItem();
    onGetItemPage(data: ItemPageModel): void;
    onGetItemAccessibility(data: AccResourceGroupModel[]): void;
    onError(err: string): void;
    private getItemPage();
    private getAboutItem();
    private getItemAccessibility();
    onSave: (newGroups: AccResourceGroupModel[]) => void;
    updateIsaapHandler(resources: AccResourceGroupModel[]): void;
    updateIsaapCookieHandler(accGroups?: AccResourceGroupModel[]): void;
    onReset: () => void;
    fetchUpdatedAboutThisItem(): void;
    onFetchedUpdatedViewModel(viewModel: AboutItemModel): void;
    onFetchUpdatedAboutError(err: string): void;
    render(): JSX.Element;
}
