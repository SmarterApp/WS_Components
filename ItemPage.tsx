import * as React from 'react'
import { QuestionTableData } from "../Models";
import * as ItemTable from './QuestionDataTable';

interface Props {
    tableData: QuestionTableData;
    title: string
}

export class Component extends React.Component<Props, undefined> {
    render() {
        return (
            <div className='item-page'>
                <ItemTable.Component tableData={this.props.tableData} />
            </div>
        );
    }
}