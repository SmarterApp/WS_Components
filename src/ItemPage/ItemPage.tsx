/// <reference types="google.analytics" />
import "../Styles/item.less";
import * as React from "react";
import * as ReactDOM from "react-dom";
import * as Accessibility from "../Accessibility/AccessibilityModels";
import * as AccessibilityModal from "../Accessibility/AccessibilityModal";
import * as Dropdown from "../DropDown/DropDown";
import { MoreLikeThisModal } from "../Modals/MoreLikeThisModal";
import { AboutItem } from "../AboutItem/AboutItem";
import { AboutPTModal } from "../PerformanceType/AboutPT";
import { AboutPTPopupModal } from "../PerformanceType/AboutPTPopup";
import * as Braille from "../Accessibility/Braille";
import { ShareModal } from "../Modals/ShareModal";
import * as ItemPageModels from "./ItemPageModels";
import { ItemViewerFrame } from "../ItemViewer/ItemViewerFrame";
import * as $ from "jquery";
import { AboutItemModel } from "../AboutItem/AboutItemModels";
import {
  AccResourceGroupModel,
  ResourceSelectionsModel
} from "../Accessibility/AccessibilityModels";
import { ItemAccessibilityModal } from "../Accessibility/AccessibilityModal";

export interface ItemPageProps extends ItemPageModels.ItemPageModel {
  onSave: (selections: ResourceSelectionsModel) => void;
  onReset: () => void;
  aboutThisItemVM: AboutItemModel;
  currentItem: ItemPageModels.ItemIdentifierModel;
  accResourceGroups: AccResourceGroupModel[];
}

export class ItemPage extends React.Component<ItemPageProps, {}> {
  constructor(props: ItemPageProps) {
    super(props);
  }

  saveOptions = (resourceSelections: ResourceSelectionsModel): void => {
    this.props.onSave(resourceSelections);
  };

  openAboutItemModal(e: React.KeyboardEvent<HTMLAnchorElement>) {
    if (e.keyCode === 13 || e.keyCode === 23 || e.keyCode === 32) {
      const modal: any = $("#about-item-modal-container");
      modal.modal();
    }
  }

  openMoreLikeThisModal(e: React.KeyboardEvent<HTMLAnchorElement>) {
    if (e.keyCode === 13 || e.keyCode === 23 || e.keyCode === 32) {
      const modal: any = $("#more-like-this-modal-container");
      modal.modal();
    }
  }

  openShareModal(e: React.KeyboardEvent<HTMLAnchorElement>) {
    if (e.keyCode === 13 || e.keyCode === 23 || e.keyCode === 32) {
      const modal: any = $("#share-modal-container");
      modal.modal();
    }
  }

  openPerfTaskModal(e: React.KeyboardEvent<HTMLAnchorElement>) {
    if (e.keyCode === 13 || e.keyCode === 23 || e.keyCode === 32) {
      const modal: any = $("#about-performance-tasks-modal-container");
      modal.modal();
    }
  }

  openAccessibilityModal(e: React.KeyboardEvent<HTMLButtonElement>) {
    if (e.keyCode === 13 || e.keyCode === 23 || e.keyCode === 32) {
      const modal: any = $("#accessibility-modal-container");
      modal.modal();
    }
  }

  renderPerformanceItemModalBtn = (isPerformanceItem: boolean) => {
    if (!isPerformanceItem) {
      return undefined;
    }

    let btnText = (
      <span>
        <span className="item-nav-long-label">This is a </span>
        <b>Performance Task</b>
      </span>
    );

    return (
      <a
        className="item-nav-btn"
        data-toggle="modal"
        data-target="#about-performance-tasks-modal-container"
        onKeyUp={e => this.openPerfTaskModal(e)}
        role="button"
        tabIndex={0}
      >
        <span
          className="glyphicon glyphicon-info-sign glyphicon-pad"
          aria-hidden="true"
        />
        {btnText}
      </a>
    );
  };

  renderModals(): JSX.Element {
    const abtText = (
      <span>
        About <span className="item-nav-long-label">This Item</span>
      </span>
    );
    const moreText = (
      <span>
        More <span className="item-nav-long-label">Like This</span>
      </span>
    );
    //TODO:Refactor modals.

    return (
      <div
        className="item-nav"
        role="toolbar"
        aria-label="Toolbar with button groups"
      >
        <div
          className="item-nav-left-group"
          role="group"
          aria-label="First group"
        >
          <a
            className="item-nav-btn"
            data-toggle="modal"
            data-target="#about-item-modal-container"
            onKeyUp={e => this.openAboutItemModal(e)}
            role="button"
            tabIndex={0}
          >
            <span
              className="glyphicon glyphicon-info-sign glyphicon-pad"
              aria-hidden="true"
            />
            {abtText}
          </a>
          <a
            className="item-nav-btn"
            data-toggle="modal"
            data-target="#more-like-this-modal-container"
            onKeyUp={e => this.openMoreLikeThisModal(e)}
            role="button"
            tabIndex={0}
          >
            <span
              className="glyphicon glyphicon-th-large glyphicon-pad"
              aria-hidden="true"
            />
            {moreText}
          </a>
          <a
            className="item-nav-btn"
            data-toggle="modal"
            data-target="#share-modal-container"
            onKeyUp={e => this.openShareModal(e)}
            role="button"
            tabIndex={0}
          >
            <span
              className="glyphicon glyphicon-share-alt glyphicon-pad"
              aria-hidden="true"
            />
            Share
          </a>
          {this.renderPerformanceItemModalBtn(this.props.isPerformanceItem)}
          <Braille.BrailleLink
            currentSelectionCode={Accessibility.getBrailleAccommodation(
              this.props.accResourceGroups
            )}
            brailleItemCodes={this.props.brailleItemCodes}
            braillePassageCodes={this.props.braillePassageCodes}
            bankKey={this.props.currentItem.bankKey}
            itemKey={this.props.currentItem.itemKey}
          />
        </div>
        {this.renderModalFrames()}
        {this.renderAccessabilityBtn()}
      </div>
    );
  }

  renderAccessabilityBtn(): JSX.Element {
    return (
      <div
        className="item-nav-right-group"
        role="group"
        aria-label="Second group"
      >
        <button
          className="accessibility-btn btn btn-primary"
          data-toggle="modal"
          data-target="#accessibility-modal-container"
          onClick={e => ga("send", "event", "button", "OpenAccessibility")}
          onKeyUp={e => this.openAccessibilityModal(e)}
          tabIndex={0}
        >
          <span
            className="glyphicon glyphicon-collapse-down"
            aria-hidden="true"
          />
          Accessibility
        </button>
      </div>
    );
  }

  renderModalFrames(): JSX.Element {
    let isaap = ItemPageModels.toiSAAP(
      this.props.accResourceGroups,
      this.props.defaultIsaapCodes
    );
    return (
      <div>
        <AboutItem {...this.props.aboutThisItemVM} />
        <ItemAccessibilityModal
          accResourceGroups={this.props.accResourceGroups}
          onSave={this.props.onSave}
          onReset={this.props.onReset}
        />
        <MoreLikeThisModal {...this.props.moreLikeThisVM} />
        <ShareModal iSAAP={isaap} />
        <AboutPTPopupModal
          subject={this.props.subject}
          description={this.props.performanceItemDescription}
          isPerformance={this.props.isPerformanceItem}
        />
        <AboutPTModal
          subject={this.props.subject}
          description={this.props.performanceItemDescription}
        />
      </div>
    );
  }

  renderIFrame(): JSX.Element {
    let isaap = ItemPageModels.toiSAAP(
      this.props.accResourceGroups,
      this.props.defaultIsaapCodes
    );
    const itemNames = Accessibility.isBrailleEnabled(
      this.props.accResourceGroups
    )
      ? this.props.brailleItemNames
      : this.props.itemNames;
    let scrollTo: string = Accessibility.isStreamlinedEnabled(
      this.props.accResourceGroups
    )
      ? ""
      : "&scrollToId=".concat(this.props.currentItem.itemName);
    let ivsUrl: string = this.props.itemViewerServiceUrl.concat(
      "/items?ids=",
      itemNames,
      "&isaap=",
      isaap,
      scrollTo
    );

    return <ItemViewerFrame url={ivsUrl} />;
  }

  render() {
    return (
      <div>
        {this.renderModals()}
        {this.renderIFrame()}
      </div>
    );
  }
}
