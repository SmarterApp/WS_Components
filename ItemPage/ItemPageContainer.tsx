import '../Styles/item.less';
import * as React from 'react';
import * as ReactDOM from 'react-dom';
import * as Accessibility from '../Accessibility/Accessibility';
import * as AboutThisItem from '../AboutItem/AboutThisItem';
import * as ItemPage from './ItemPage';
import * as ItemPageModels from './ItemPageModels';
import { Resource, get, getResourceContent, parseQueryString } from '../ApiModel';
import { RouteComponentProps } from 'react-router';

export const AboutThisItemViewModelClient = (params: ItemPageModels.Item) =>
    get<AboutThisItem.Props>("/Item/AboutThisItemViewModel", params);
export const ItemPageClient = (params: ItemPageModels.ItemIsaap) =>
    get<ItemPageModels.ItemPageViewModel>("/Item/GetItem", params);


interface Props extends RouteComponentProps<{}> {
    aboutThisClient: (params: ItemPageModels.Item) =>
        Promise<AboutThisItem.Props>;
    itemPageClient: (params: ItemPageModels.ItemIsaap) =>
        Promise<ItemPageModels.ItemPageViewModel>;
}

interface State {
    aboutThisItem: Resource<AboutThisItem.Props>
    itemPageVM: Resource<ItemPageModels.ItemPageViewModel>;
    currentItem?: ItemPageModels.ItemIdentifier;
    item: ItemPageModels.Item;

}

export class ItemPageContainer extends React.Component<Props, State>{
    constructor(props: Props) {
        super(props);
        //TODO: parse url params and change item

        const queryObject = parseQueryString(location.search);
        const itemKey = +(queryObject["itemKey"] || [])[0] || 0;
        const bankKey = +(queryObject["bankKey"] || [])[0] || 0;
        const isaap = (queryObject["isaap"] || [])[0] || "";
        
        const item: ItemPageModels.ItemIsaap = { itemKey: itemKey, bankKey: bankKey, isaap: isaap }

        this.state = {
            aboutThisItem: { kind: "loading" },
            itemPageVM: { kind: "loading" },
            item: item
        }

        this.props.itemPageClient(this.state.item)
            .then((data) => this.onGetItemPage(data))
            .then(() => this.setCurrentItem())
            .then(() => this.fetchUpdatedAboutThisItem())
            .catch(err => this.onError(err));

    }

    private setCurrentItem() {
        const itemPage = this.getItemPage();
        let currentItem: ItemPageModels.ItemIdentifier | undefined = undefined;
        if (itemPage) {
            currentItem = Accessibility.isBrailleEnabled(itemPage.accResourceGroups) ?
                itemPage.brailleItem : itemPage.nonBrailleItem;
        }

        this.setState({
            currentItem: currentItem
        });

    }

    onGetItemPage(data: ItemPageModels.ItemPageViewModel) {
        this.setState({
            itemPageVM: { kind: "success", content: data }
        });

    }

    onError(err: any) {
        console.error(err);
    }

    private getItemPage(): ItemPageModels.ItemPageViewModel | undefined {
        const itemPage = this.state.itemPageVM;
        return getResourceContent(itemPage);
    }

    private getAboutItem(): AboutThisItem.Props | undefined {
        const aboutItem = this.state.aboutThisItem;
        return getResourceContent(aboutItem);
    }



    onSave = (selections: Accessibility.ResourceSelections) => {
        const itemPage = this.getItemPage();
        if (itemPage) {

            const newGroups: Accessibility.AccResourceGroup[] = [];
            for (let group of itemPage.accResourceGroups) {
                const newGroup = { ...group };
                const newResources: Accessibility.AccessibilityResource[] = [];
                for (let res of newGroup.accessibilityResources) {
                    const newRes = { ...res };
                    newRes.currentSelectionCode = selections[newRes.resourceCode] || newRes.currentSelectionCode;
                    newResources.push(newRes);
                }
                newGroup.accessibilityResources = newResources;
                newGroups.push(newGroup);
            }

            const newItemPage = { ...itemPage };
            newItemPage.accResourceGroups = newGroups;

            this.onGetItemPage(newItemPage);
            this.setCurrentItem()
            this.updateCookie(newItemPage.accessibilityCookieName, newItemPage.accResourceGroups);

        }
        else {
            console.error('Error no item to update resources');
        }

    }

    private updateCookie(cookieName: string, accGroups?: Accessibility.AccResourceGroup[]): void {
        let cookieValue = (accGroups) ? ItemPageModels.toCookie(accGroups) : "";
        document.cookie = cookieName.concat("=", cookieValue, "; path=/");
    }

    onReset = () => {
        const itemPage = this.getItemPage();

        if (itemPage) {
            const newItemPage = { ...itemPage };

            this.updateCookie(newItemPage.accessibilityCookieName);

            const newAccResourceGroups = newItemPage.accResourceGroups.map(g => {
                const newGroup = { ...g };
                newGroup.accessibilityResources = newGroup.accessibilityResources.map(ItemPageModels.resetResource);
                return newGroup;
            });

            newItemPage.accResourceGroups = newAccResourceGroups;
            this.onGetItemPage(newItemPage);
            this.setCurrentItem();
            this.fetchUpdatedAboutThisItem();

        }
        else {
            console.error('Error no item to reset');
        }

    }

    fetchUpdatedAboutThisItem() {
        const item = this.state.currentItem;
        if (item) {
            const params = {
                bankKey: item.bankKey,
                itemKey: item.itemKey
            };
            AboutThisItemViewModelClient(params).then((data) => this.onFetchedUpdatedViewModel(data)).catch();
        }

    }

    onFetchedUpdatedViewModel(viewModel: AboutThisItem.Props) {
        this.setState({
            aboutThisItem: { kind: "success", content: viewModel }
        });
    }

    onFetchUpdatedAboutError(err: any) {
        console.error(err);
        this.setState({
            aboutThisItem: { kind: "failure" }
        });
    }

    render() {
        const aboutThisItem = this.getAboutItem();
        const itemDetails = this.state.currentItem;
        const itemPage = this.getItemPage();

        if (aboutThisItem && itemPage && itemDetails) {
            return <div className="item-page">
                <ItemPage.Page
                    {...itemPage}
                    aboutThisItemVM={aboutThisItem}
                    onSave={this.onSave}
                    onReset={this.onReset}
                    currentItem={itemDetails}
                />
            </div>
        }
        else {
            return <div></div>
        }
    }
}

