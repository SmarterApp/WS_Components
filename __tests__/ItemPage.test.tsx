import * as React from 'react';
import * as ReactDOM from 'react-dom';
import { ItemPage } from '../ItemPage';
import * as Models from "../../Models";
import { shallow } from 'enzyme';
import { GradeLevels } from '../../../../client/src/Models/GradeLevels';


const rubric:  Models.Rubric = {
    language: "english",
    rubricEntries: [{    
        scorepoint: "",
        name: "",
        value: ""
    }],
    samples: [{
        maxValue: "",
        minValue: "",
        sampleResponses: [{
            purpose: "",
            scorePoint: "",
            name: "",
            sampleContent: ""
        }]     
    }]
}

const itemVM: Models.ItemViewModel = {
    bankKey: 1,
    itemKey: 1,
    gradeLabel: "grade 3",
    subjectLabel: "math",
    claimCode: "234536",
    targetId: "23463467",
    domain: "MATH",
    depthOfKnowledge: "",
    commonCoreStandardId: "",
    title: "",
    grade: GradeLevels.Grade6,
    subjectCode: "MATH",
    claimLabel: "Math Claim",
    target: "1-3",
    interactionTypeCode: "",
    interactionTypeLabel: "",
    isPerformanceItem: false
}

const aboutItemVM: Models.AboutItemViewModel = {
    itemCardViewModel: itemVM,
    rubrics: [rubric],
    targetDescription: "string",
    depthOfKnowledge: "string",
    commonCoreStandardsDescription: "string",
    educationalDifficulty: "string",
    evidenceStatement: "string",
    associatedItems: "string"
}

const itemView: Models.ItemView = {
    id: "12334",
    html: null,
    picturePath: null,
    captured: true,
    type: Models.ViewType.html
}

const question: Models.Question = {
    id: "12334",
    view: itemView,
    data: aboutItemVM,
    questionNumber: 1
}

const itemGroup : Models.ItemGroup = {
    passage: null,
    questions: [question]
}

describe("EvidenceStatement", () => {
    it("matches snapshot", () => {
        expect(shallow(<ItemPage itemData={itemGroup} />)).toMatchSnapshot();
    })
})
