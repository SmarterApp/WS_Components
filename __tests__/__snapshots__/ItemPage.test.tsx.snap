// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`EvidenceStatement matches snapshot 1`] = `
ShallowWrapper {
  "complexSelector": ComplexSelector {
    "buildPredicate": [Function],
    "childrenOfNode": [Function],
    "findWhereUnwrapped": [Function],
  },
  "length": 1,
  "node": <div
    className="page"
  >
    <QuestionView
      question={
        Object {
          "data": Object {
            "associatedItems": "string",
            "commonCoreStandardsDescription": "string",
            "depthOfKnowledge": "string",
            "educationalDifficulty": "string",
            "evidenceStatement": "string",
            "itemCardViewModel": Object {
              "bankKey": 1,
              "claimCode": "234536",
              "claimLabel": "Math Claim",
              "commonCoreStandardId": "",
              "depthOfKnowledge": "",
              "domain": "MATH",
              "grade": 8,
              "gradeLabel": "grade 3",
              "interactionTypeCode": "",
              "interactionTypeLabel": "",
              "isPerformanceItem": false,
              "itemKey": 1,
              "subjectCode": "MATH",
              "subjectLabel": "math",
              "target": "1-3",
              "targetId": "23463467",
              "title": "",
            },
            "rubrics": Array [
              Object {
                "language": "english",
                "rubricEntries": Array [
                  Object {
                    "name": "",
                    "scorepoint": "",
                    "value": "",
                  },
                ],
                "samples": Array [
                  Object {
                    "maxValue": "",
                    "minValue": "",
                    "sampleResponses": Array [
                      Object {
                        "name": "",
                        "purpose": "",
                        "sampleContent": "",
                        "scorePoint": "",
                      },
                    ],
                  },
                ],
              },
            ],
            "targetDescription": "string",
          },
          "id": "12334",
          "questionNumber": 1,
          "view": Object {
            "captured": true,
            "html": null,
            "id": "12334",
            "picturePath": null,
            "type": 1,
          },
        }
      }
    />
  </div>,
  "nodes": Array [
    <div
      className="page"
    >
      <QuestionView
        question={
          Object {
            "data": Object {
              "associatedItems": "string",
              "commonCoreStandardsDescription": "string",
              "depthOfKnowledge": "string",
              "educationalDifficulty": "string",
              "evidenceStatement": "string",
              "itemCardViewModel": Object {
                "bankKey": 1,
                "claimCode": "234536",
                "claimLabel": "Math Claim",
                "commonCoreStandardId": "",
                "depthOfKnowledge": "",
                "domain": "MATH",
                "grade": 8,
                "gradeLabel": "grade 3",
                "interactionTypeCode": "",
                "interactionTypeLabel": "",
                "isPerformanceItem": false,
                "itemKey": 1,
                "subjectCode": "MATH",
                "subjectLabel": "math",
                "target": "1-3",
                "targetId": "23463467",
                "title": "",
              },
              "rubrics": Array [
                Object {
                  "language": "english",
                  "rubricEntries": Array [
                    Object {
                      "name": "",
                      "scorepoint": "",
                      "value": "",
                    },
                  ],
                  "samples": Array [
                    Object {
                      "maxValue": "",
                      "minValue": "",
                      "sampleResponses": Array [
                        Object {
                          "name": "",
                          "purpose": "",
                          "sampleContent": "",
                          "scorePoint": "",
                        },
                      ],
                    },
                  ],
                },
              ],
              "targetDescription": "string",
            },
            "id": "12334",
            "questionNumber": 1,
            "view": Object {
              "captured": true,
              "html": null,
              "id": "12334",
              "picturePath": null,
              "type": 1,
            },
          }
        }
      />
    </div>,
  ],
  "options": Object {},
  "renderer": ReactShallowRenderer {
    "_instance": ShallowComponentWrapper {
      "_calledComponentWillUnmount": false,
      "_compositeType": 0,
      "_context": Object {},
      "_currentElement": <ItemPage
        itemData={
          Object {
            "passage": null,
            "questions": Array [
              Object {
                "data": Object {
                  "associatedItems": "string",
                  "commonCoreStandardsDescription": "string",
                  "depthOfKnowledge": "string",
                  "educationalDifficulty": "string",
                  "evidenceStatement": "string",
                  "itemCardViewModel": Object {
                    "bankKey": 1,
                    "claimCode": "234536",
                    "claimLabel": "Math Claim",
                    "commonCoreStandardId": "",
                    "depthOfKnowledge": "",
                    "domain": "MATH",
                    "grade": 8,
                    "gradeLabel": "grade 3",
                    "interactionTypeCode": "",
                    "interactionTypeLabel": "",
                    "isPerformanceItem": false,
                    "itemKey": 1,
                    "subjectCode": "MATH",
                    "subjectLabel": "math",
                    "target": "1-3",
                    "targetId": "23463467",
                    "title": "",
                  },
                  "rubrics": Array [
                    Object {
                      "language": "english",
                      "rubricEntries": Array [
                        Object {
                          "name": "",
                          "scorepoint": "",
                          "value": "",
                        },
                      ],
                      "samples": Array [
                        Object {
                          "maxValue": "",
                          "minValue": "",
                          "sampleResponses": Array [
                            Object {
                              "name": "",
                              "purpose": "",
                              "sampleContent": "",
                              "scorePoint": "",
                            },
                          ],
                        },
                      ],
                    },
                  ],
                  "targetDescription": "string",
                },
                "id": "12334",
                "questionNumber": 1,
                "view": Object {
                  "captured": true,
                  "html": null,
                  "id": "12334",
                  "picturePath": null,
                  "type": 1,
                },
              },
            ],
          }
        }
      />,
      "_debugID": 1,
      "_hostContainerInfo": null,
      "_hostParent": null,
      "_instance": ItemPage {
        "_reactInternalInstance": [Circular],
        "context": Object {},
        "props": Object {
          "itemData": Object {
            "passage": null,
            "questions": Array [
              Object {
                "data": Object {
                  "associatedItems": "string",
                  "commonCoreStandardsDescription": "string",
                  "depthOfKnowledge": "string",
                  "educationalDifficulty": "string",
                  "evidenceStatement": "string",
                  "itemCardViewModel": Object {
                    "bankKey": 1,
                    "claimCode": "234536",
                    "claimLabel": "Math Claim",
                    "commonCoreStandardId": "",
                    "depthOfKnowledge": "",
                    "domain": "MATH",
                    "grade": 8,
                    "gradeLabel": "grade 3",
                    "interactionTypeCode": "",
                    "interactionTypeLabel": "",
                    "isPerformanceItem": false,
                    "itemKey": 1,
                    "subjectCode": "MATH",
                    "subjectLabel": "math",
                    "target": "1-3",
                    "targetId": "23463467",
                    "title": "",
                  },
                  "rubrics": Array [
                    Object {
                      "language": "english",
                      "rubricEntries": Array [
                        Object {
                          "name": "",
                          "scorepoint": "",
                          "value": "",
                        },
                      ],
                      "samples": Array [
                        Object {
                          "maxValue": "",
                          "minValue": "",
                          "sampleResponses": Array [
                            Object {
                              "name": "",
                              "purpose": "",
                              "sampleContent": "",
                              "scorePoint": "",
                            },
                          ],
                        },
                      ],
                    },
                  ],
                  "targetDescription": "string",
                },
                "id": "12334",
                "questionNumber": 1,
                "view": Object {
                  "captured": true,
                  "html": null,
                  "id": "12334",
                  "picturePath": null,
                  "type": 1,
                },
              },
            ],
          },
        },
        "refs": Object {},
        "state": null,
        "updater": Object {
          "enqueueCallback": [Function],
          "enqueueCallbackInternal": [Function],
          "enqueueElementInternal": [Function],
          "enqueueForceUpdate": [Function],
          "enqueueReplaceState": [Function],
          "enqueueSetState": [Function],
          "isMounted": [Function],
          "validateCallback": [Function],
        },
      },
      "_mountOrder": 1,
      "_pendingCallbacks": null,
      "_pendingElement": null,
      "_pendingForceUpdate": false,
      "_pendingReplaceState": false,
      "_pendingStateQueue": null,
      "_renderedComponent": NoopInternalComponent {
        "_currentElement": <div
          className="page"
        >
          <QuestionView
            question={
              Object {
                "data": Object {
                  "associatedItems": "string",
                  "commonCoreStandardsDescription": "string",
                  "depthOfKnowledge": "string",
                  "educationalDifficulty": "string",
                  "evidenceStatement": "string",
                  "itemCardViewModel": Object {
                    "bankKey": 1,
                    "claimCode": "234536",
                    "claimLabel": "Math Claim",
                    "commonCoreStandardId": "",
                    "depthOfKnowledge": "",
                    "domain": "MATH",
                    "grade": 8,
                    "gradeLabel": "grade 3",
                    "interactionTypeCode": "",
                    "interactionTypeLabel": "",
                    "isPerformanceItem": false,
                    "itemKey": 1,
                    "subjectCode": "MATH",
                    "subjectLabel": "math",
                    "target": "1-3",
                    "targetId": "23463467",
                    "title": "",
                  },
                  "rubrics": Array [
                    Object {
                      "language": "english",
                      "rubricEntries": Array [
                        Object {
                          "name": "",
                          "scorepoint": "",
                          "value": "",
                        },
                      ],
                      "samples": Array [
                        Object {
                          "maxValue": "",
                          "minValue": "",
                          "sampleResponses": Array [
                            Object {
                              "name": "",
                              "purpose": "",
                              "sampleContent": "",
                              "scorePoint": "",
                            },
                          ],
                        },
                      ],
                    },
                  ],
                  "targetDescription": "string",
                },
                "id": "12334",
                "questionNumber": 1,
                "view": Object {
                  "captured": true,
                  "html": null,
                  "id": "12334",
                  "picturePath": null,
                  "type": 1,
                },
              }
            }
          />
        </div>,
        "_debugID": 2,
        "_renderedOutput": <div
          className="page"
        >
          <QuestionView
            question={
              Object {
                "data": Object {
                  "associatedItems": "string",
                  "commonCoreStandardsDescription": "string",
                  "depthOfKnowledge": "string",
                  "educationalDifficulty": "string",
                  "evidenceStatement": "string",
                  "itemCardViewModel": Object {
                    "bankKey": 1,
                    "claimCode": "234536",
                    "claimLabel": "Math Claim",
                    "commonCoreStandardId": "",
                    "depthOfKnowledge": "",
                    "domain": "MATH",
                    "grade": 8,
                    "gradeLabel": "grade 3",
                    "interactionTypeCode": "",
                    "interactionTypeLabel": "",
                    "isPerformanceItem": false,
                    "itemKey": 1,
                    "subjectCode": "MATH",
                    "subjectLabel": "math",
                    "target": "1-3",
                    "targetId": "23463467",
                    "title": "",
                  },
                  "rubrics": Array [
                    Object {
                      "language": "english",
                      "rubricEntries": Array [
                        Object {
                          "name": "",
                          "scorepoint": "",
                          "value": "",
                        },
                      ],
                      "samples": Array [
                        Object {
                          "maxValue": "",
                          "minValue": "",
                          "sampleResponses": Array [
                            Object {
                              "name": "",
                              "purpose": "",
                              "sampleContent": "",
                              "scorePoint": "",
                            },
                          ],
                        },
                      ],
                    },
                  ],
                  "targetDescription": "string",
                },
                "id": "12334",
                "questionNumber": 1,
                "view": Object {
                  "captured": true,
                  "html": null,
                  "id": "12334",
                  "picturePath": null,
                  "type": 1,
                },
              }
            }
          />
        </div>,
      },
      "_renderedNodeType": 0,
      "_rootNodeID": 0,
      "_topLevelWrapper": null,
      "_updateBatchNumber": null,
      "_warnedAboutRefsInRender": false,
    },
    "getRenderOutput": [Function],
    "render": [Function],
  },
  "root": [Circular],
  "unrendered": <ItemPage
    itemData={
      Object {
        "passage": null,
        "questions": Array [
          Object {
            "data": Object {
              "associatedItems": "string",
              "commonCoreStandardsDescription": "string",
              "depthOfKnowledge": "string",
              "educationalDifficulty": "string",
              "evidenceStatement": "string",
              "itemCardViewModel": Object {
                "bankKey": 1,
                "claimCode": "234536",
                "claimLabel": "Math Claim",
                "commonCoreStandardId": "",
                "depthOfKnowledge": "",
                "domain": "MATH",
                "grade": 8,
                "gradeLabel": "grade 3",
                "interactionTypeCode": "",
                "interactionTypeLabel": "",
                "isPerformanceItem": false,
                "itemKey": 1,
                "subjectCode": "MATH",
                "subjectLabel": "math",
                "target": "1-3",
                "targetId": "23463467",
                "title": "",
              },
              "rubrics": Array [
                Object {
                  "language": "english",
                  "rubricEntries": Array [
                    Object {
                      "name": "",
                      "scorepoint": "",
                      "value": "",
                    },
                  ],
                  "samples": Array [
                    Object {
                      "maxValue": "",
                      "minValue": "",
                      "sampleResponses": Array [
                        Object {
                          "name": "",
                          "purpose": "",
                          "sampleContent": "",
                          "scorePoint": "",
                        },
                      ],
                    },
                  ],
                },
              ],
              "targetDescription": "string",
            },
            "id": "12334",
            "questionNumber": 1,
            "view": Object {
              "captured": true,
              "html": null,
              "id": "12334",
              "picturePath": null,
              "type": 1,
            },
          },
        ],
      }
    }
  />,
}
`;
